### DISCUSSION TITLE
"module is not iterable" error on one of my repos

### DISCUSSION THREAD

**Discussion by toestreich** [renovatebot/renovate#20182] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

v34.109.1 of the Gothub Action (uses v34.109.1-slim)

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

Completed renovate scan 

### What you tried so far.

Changed the version of the github action to prior builds and got an Error removing dangling containers error prior to the fatal one below (similar to this issue https://github.com/renovatebot/docker-renovate-full/issues/9)

### Relevant debug logs

<details><summary>Logs</summary>

```
Status: Downloaded newer image for renovate/renovate:34.109.1-slim
 INFO: Repository started (repository=ORG/REPO)
       "renovateVersion": "34.109.1"
ERROR: Repository has unknown error (repository=ORG/REPO)
       "err": {
         "message": "module is not iterable",
         "stack": "TypeError: module is not iterable\n    at ModuleExtractor.extract (/opt/buildpack/tools/renovate/34.109.1/node_modules/renovate/lib/modules/manager/terraform/extractors/others/modules.ts:39:35)\n    at Object.extractPackageFile (/opt/buildpack/tools/renovate/34.109.1/node_modules/renovate/lib/modules/manager/terraform/extract.ts:49:28)\n    at getManagerPackageFiles (/opt/buildpack/tools/renovate/34.109.1/node_modules/renovate/lib/workers/repository/extract/manager-files.ts:52:19)\n    at /opt/buildpack/tools/renovate/34.109.1/node_modules/renovate/lib/workers/repository/extract/index.ts:47:28\n    at async Promise.all (index 7)\n    at extractAllDependencies (/opt/buildpack/tools/renovate/34.109.1/node_modules/renovate/lib/workers/repository/extract/index.ts:45:26)\n    at extract (/opt/buildpack/tools/renovate/34.109.1/node_modules/renovate/lib/workers/repository/process/extract-update.ts:117:20)\n    at extractDependencies (/opt/buildpack/tools/renovate/34.109.1/node_modules/renovate/lib/workers/repository/process/index.ts:120:26)\n    at Object.renovateRepository (/opt/buildpack/tools/renovate/34.109.1/node_modules/renovate/lib/workers/repository/index.ts:56:9)\n    at attributes.repository (/opt/buildpack/tools/renovate/34.109.1/node_modules/renovate/lib/workers/global/index.ts:181:11)\n    at start (/opt/buildpack/tools/renovate/34.109.1/node_modules/renovate/lib/workers/global/index.ts:166:7)\n    at /opt/buildpack/tools/renovate/34.109.1/node_modules/renovate/lib/renovate.ts:18:22"```

</details>

**Comment by rarkins** [comment]
Can you reproduce this problem in a public repo?

Can you look at the affected repo for anything unusual about that file? It looks like it's related to a "modules" field somewhere. DEBUG logs could help here too.

**Reply by toestreich** [reply]
I will try to replicate - i think the error is referring to the extractor modules introduced in [this release](https://github.com/renovatebot/renovate/compare/34.12.0..34.109.1?diff=split)?

**Comment by toestreich** [comment]
@rarkins  I am unable to replicate, however the issue is still ongoing in our repo. Is there any way to narrow down the package or process in this particular repo that is causing this bug? I believe the "module" the error referring to is the module extractor module within renovate (lib/modules/manager/terraform/extractors/others/modules.spec.ts on line 10)

**Reply by rarkins** [reply]
If it's a private repo then probably only you can

**Reply by viceice** [reply]
this function should have a try catch
https://github.com/renovatebot/renovate/blob/8a08c8916bfc30ffed63f0a1ff81a968af81f10e/lib/modules/manager/terraform/extract.ts#L11

**Reply by viceice** [reply]
and here we fail
https://github.com/renovatebot/renovate/blob/8a08c8916bfc30ffed63f0a1ff81a968af81f10e/lib/modules/manager/terraform/extractors/others/modules.ts#L38

so the check here isn't enough 
https://github.com/renovatebot/renovate/blob/8a08c8916bfc30ffed63f0a1ff81a968af81f10e/lib/modules/manager/terraform/extractors/others/modules.ts#L33

**Reply by viceice** [reply]
@secustor @zharinov do you have some time to make those changes?

**Reply by zharinov** [reply]
I'll do

**Reply by zharinov** [reply]
https://github.com/renovatebot/renovate/pull/20394
